ci/build.ps1

```diff
@@ -35,95 +45,131 @@ if ($compiler -eq 'MINGW') {
     $nvimCmakeVars['USE_GCOV'] = 'ON'
     $uploadToCodecov = $true
   }
+
   # These are native MinGW builds, but they use the toolchain inside
   # MSYS2, this allows using all the dependencies and tools available
   # in MSYS2, but we cannot build inside the MSYS2 shell.
   $cmakeGenerator = 'MinGW Makefiles'
-  $cmakeGeneratorArgs = 'VERBOSE=1'
 
-  # Add MinGW to the PATH
-  $env:PATH = "C:\msys64\mingw$bits\bin;$env:PATH"
+  if ($cmakeGenerator -eq 'Ninja') {
+    $cmakeGeneratorArgs = @('-v')
+  }
+  elseif ($cmakeGenerator -eq 'MinGW Makefiles') {
+    $env:CC = 'cc'
+    $cmakeGeneratorArgs = @('VERBOSE=1')
+    @{
+      CMAKE_SH = 'CMAKE_SH-NOTFOUND'
+      CMAKE_MAKE_PROGRAM = "C:/msys64/mingw$bits/bin/mingw32-make.exe"
+    }.GetEnumerator() | ForEach-Object {
+      $nvimCmakeVars[$_.Key] = $_.Value
+      $depsCmakeVars[$_.Key] = $_.Value
+      $cmakeGeneratorArgs += $_.Key + '=' + $_.Value
+    }
+  }
 
   # Build third-party dependencies
+  $mingwPackages = @('gcc', 'make', 'cmake', 'perl', 'diffutils', 'unibilium').ForEach({
+    @('mingw', 'w64', $arch, $_) -join '-'
```

Comment:
The old code was more obvious. Is there a functional need for this change?

Reply:
No. I was testing the pattern so I can conveniently add gcc, make, ninja. I prefer joining arrays than `'mingw-w64-' + $arch + '-' + $_` and all packages had the same naming pattern.